pyproject.toml

```
@@ -49,11 +49,75 @@ line_length = 100
 combine_as_imports = true
 filter_files = true
 
+[tool.ruff]
+line-length = 100
+target-version = "py38"
+src = ["disnake"]
+select = [
+    # commented out codes are intended to be enabled in future prs
+    "F", # pyflakes
+    "E","W", # pycodestyle
+    # "D", # pydocstyle
+    # "ANN", # flake8-annotations
+    "S", # flake8-bandit
+    "B", # flake8-bugbear
+    "C", # flake8-comprehensions
+    # "DTZ", # flake8-datetimez
+    # "EM", # flake8-errmsg
+    "G", # flake8-logging-format
+    # "RET", # flake8-return
+    # "SIM", # flake8-simplify
+    "TID251", # flake8-tidy-imports, replaces S404
+    # "TCH", # flake8-type-checking
+    "RUF", # ruff specific exceptions
+    "PT", # flake8-pytest-style
+    "Q", # flake8-quotes
+    "T20", # flake8-print
+    "PGH", # pygrep-hooks
+    "TRY002", "TRY004", # tryceratops
+]
+ignore = [
+    # star imports
+    "F401","F403",
+
+    # unknown if this is actually an issue
+    "RUF006",
+
+    # ignore try-except-pass. Bare excepts are caught with E722
+    "S110",
+
+    # provide specific codes on type: ignore
+    "PGH003",
+
+    # temporary disables, to fix later
+    "B904", # within an except clause raise from error or from none
+    "B026", # backwards star-arg unpacking
+    "E501", # line too long
+    "E731", # assigning lambdas to variables
+    "RUF004", # keyword args must be supplied after starred arguments
+    "T201", # print statements
+]
+
+[tool.ruff.per-file-ignores]
+"disnake/__main__.py" = ["T201"]
+"disnake/i18n.py" = ["B027"] # lib bug
+"disnake/**.py" = ["PT"]
+"examples/*.py" = ["B008", "T201", "PT"]
+"examples/basic_voice.py" = ["S104"]
+"examples/views/tic_tac_toe.py" = ["E741", "RET505"]
+"test_bot/*.py" = ["B008", "T201", "PT"]
+"tests/*.py" = ["S101"]
+
+[tool.ruff.flake8-pytest-style]
+fixture-parentheses = false
+mark-parentheses = false
+
+[tool.ruff.flake8-tidy-imports.banned-api]
+"subprocess".msg = "Consider possible security implications associated with the subprocess module." # replaces S404
```

```suggestion
[tool.ruff.flake8-tidy-imports.banned-api.subprocess]
msg = "Consider possible security implications associated with the subprocess module." # replaces S404
```
This should make vscode's toml parser shut up.

---

You should use a different plugin. Mine does not complain about this.